features/selection.feature

```diff
@@ -2,60 +2,101 @@ Feature: select values using query syntax
       Given I query for "ex:favouriteReads"
       When I query the result for "so:author @value"
       Then the result should be "Iain M Banks"
-    
+
     Scenario: Query for the first item in a collection not at the top level
       When I query for "so:author @value"
       Then the result should be "Iain M Banks"
-    
+
     Scenario: Get the JSON for a selected item
       Given I query for "ex:favouriteReads"
       When I get the result's json
       Then the json should match
         | json                                                                                                                                             |
-        | [ { "http://schema.org/author": [ { "@value": "Iain M Banks" } ], "http://schema.org/name": [ { "@value": "Excession" } ], "@index": "banks-exc" }, { "http://schema.org/author": [ { "@value": "Thomas Pynchon" } ], "http://schema.org/name": [ { "@value": "Gravity's Rainbow" } ], "http://www.example.org#note-to-self": [ { "@value": "Need to finish reading this" } ], "@index": "pynchon-gr" } ] |
+        | [ { "http://schema.org/author": [ { "@value": "Iain M Banks" } ], "http://schema.org/name": [ { "@value": "Excession" } ], "@index": "banks-exc" }, { "http://schema.org/author": [ { "@value": "Iain M Banks" } ], "http://schema.org/name": [ { "@value": "The Player of Games" } ], "@index": "banks-pog" }, { "http://schema.org/author": [ { "@value": "Thomas Pynchon" } ], "http://schema.org/name": [ { "@value": "Gravity's Rainbow" } ], "http://www.example.org#note-to-self": [ { "@value": "Need to finish reading this" } ], "@index": "pynchon-gr" } ] |
 
     Scenario: Query for the author nodes
-        When I query for all "ex:favouriteReads so:author"
-        Then the result should be an array of 2 QueryNodes
-      
+      Given I query for all "ex:favouriteReads so:author"
+      Then the result should be an array of 3 QueryNodes
+
     Scenario: Query for author nodes, then for names
-        When I query for all "ex:favouriteReads so:author"
-        And then I query the result for all "@value"
-        Then the result should be an array [ "Iain M Banks" ]
+      Given I query for all "ex:favouriteReads so:author"
+      And then I query the result for all "@value"
+      Then the result should be an array [ "Iain M Banks" ]
 
-    Scenario: Query for recursive properties
\ No newline at end of file
+    Scenario: Get the first result for a specific author
+      Given I query for "ex:favouriteReads[so:author=Iain M Banks]"
+      When I get the result's json
+      Then the json should match
+        | json                                                                                                                                             |
+        | { "http://schema.org/author": [ { "@value": "Iain M Banks" } ], "http://schema.org/name": [ { "@value": "Excession" } ], "@index": "banks-exc" } |
+
+    Scenario: Query for favourite reads for a specific author
```

Comment:
This is an excellent feature, but not one which was originally intended by the concept of attributes. This is something more like an arbitrary query clause.
I suggest that you separate this out into a distinct issue. Attributes like \@index are what was intended for this feature.

Note that all features need to be more-or-less mapped to https://dev.w3.org/2006/webapi/selectors-api2/#interface-definitions